core/rend/vulkan/oit/oit_pipeline.h

```diff
@@ -162,45 +162,43 @@ class OITDescriptorSets
 			perPolyDescSet = perPolyAlloc.alloc();
 			std::vector<vk::WriteDescriptorSet> writeDescriptorSets;
 
-			vk::DescriptorImageInfo imageInfo0;
 			if (poly.texture != nullptr)
 			{
-				imageInfo0 = vk::DescriptorImageInfo{ samplerManager->GetSampler(poly.tsp), ((Texture *)poly.texture)->GetReadOnlyImageView(),
-						vk::ImageLayout::eShaderReadOnlyOptimal };
-				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, 1, vk::DescriptorType::eCombinedImageSampler, &imageInfo0, nullptr, nullptr);
+				vk::DescriptorImageInfo imageInfo0(samplerManager->GetSampler(poly.tsp), ((Texture *)poly.texture)->GetReadOnlyImageView(),
+						vk::ImageLayout::eShaderReadOnlyOptimal);
+				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, vk::DescriptorType::eCombinedImageSampler, imageInfo0);
```

Comment:
I believe there is an issue here: imageInfo0 is used by reference in the emplace_back/constructor (vk::ArrayProxyNoTemporaries) but goes out of scope before `writeDescriptorSets` is used.

Same thing for imageInfo1, uniBufferInfo and lightBufferInfo below.

This is the reason I had these variables defined outside the `if`'s

Reply:
Indeed, fixed